src/solvers/refinement/string_constraint.cpp

```diff
@@ -0,0 +1,77 @@
+/** -*- C++ -*- *****************************************************\
+
+Module: String constraints
+        (see the PASS paper at HVC'13)
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+#include <solvers/refinement/string_constraint.h>
+
+
+exprt string_constraintt::premise() const {
+  if(form == SIMPLE || form == UNIV_QUANT) {
+    if(id() == ID_implies)
+      return op0();
+    else
+      return true_exprt();
+  }
+  else {
+    return(*this);
+  }
+}
+
+exprt string_constraintt::body() const {
+  if(form == SIMPLE || form == UNIV_QUANT) {
+    if(id() == ID_implies)
+      return op1();
+    else
+      return(*this);
+  } else throw "string_constraintt::body() should not be applied to NOT_CONTAINS expression";
+}
+
+string_constraintt string_constraintt::forall(const symbol_exprt & univ, const exprt & bound_inf, const exprt & bound_sup)
+{
+  string_constraintt sc(*this);
+  sc.form = UNIV_QUANT;
+  sc.quantified_variable = univ;
+  sc.bounds.push_back(bound_inf);
+  sc.bounds.push_back(bound_sup);
+  return sc;
+}
+
+string_constraintt string_constraintt::forall(const symbol_exprt & univ, const exprt & bound_sup)
+{
+  return forall(univ,refined_string_typet::index_zero(),bound_sup);
+}
+
+string_constraintt string_constraintt::not_contains(exprt univ_bound_inf, exprt univ_bound_sup,
+				 exprt premise, exprt exists_bound_inf,
+				 exprt exists_bound_sup, exprt s0, exprt s1)
+{
+  string_constraintt sc(premise);
```

Comment:
Done in https://github.com/diffblue/cbmc/pull/341/commits/880812d989f9ee84c09d4bbc37a7863c668a4601
but this will probably end up being a constructor for its own class in another commit